esp-hal/src/dma/mod.rs

```diff
@@ -1701,46 +1658,61 @@ where
             return Err(DmaError::InvalidAlignment);
         }
 
-        self.rx_impl
-            .prepare_transfer_without_start(preparation.start, peri)
+        compiler_fence(core::sync::atomic::Ordering::SeqCst);
```

Comment:
Why do we need this fence? All operations around here are volatile, so non-reorderable

Reply:
Putting aside the "it was there before I touched it"

The fence is to make sure the descriptor and buffer setup (which is non-volatile) happens before the DMA starts.